### PR TITLE
feat(generative-ai): Add compute token examples

### PR DESCRIPTION
**Pull request by msampathkumar**
Add list_tokens_example

## Description

Fixes [b358102024](http://b/358102024)

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by msampathkumar** [comment]
CICD error `google.api_core.exceptions.ResourceExhausted: 429 Resource exhausted.`

Need to re-try after sometime.

**Comment by davidcavazos** [comment]
It's running 148 tests every time changes get pushed, which might explain the resource exhausted errors. It looks like it's running everything under `generative_ai/` because the `noxfile_config.py` is located there. To avoid running everything, try adding a `noxfile_config.py` along with the necessary `requirements.txt` and `requirements-test.txt` under `generative_ai/count_token/`. That way you can also trim down the dependencies for that specific directory.

**Comment by msampathkumar** [comment]
@davidcavazos - Thank you for the details steps. Will do as suggested. We are currently working on re-organising the samples into feature folder. [go/runnable-snippets-execute](http://go/runnable-snippets-execute)

> It's running 148 tests every time changes get pushed, which might explain the resource exhausted errors. It looks like it's running everything under `generative_ai/` because the `noxfile_config.py` is located there. To avoid running everything, try adding a `noxfile_config.py` along with the necessary `requirements.txt` and `requirements-test.txt` under `generative_ai/count_token/`. That way you can also trim down the dependencies for that specific directory.
> 
> It would be nice to do the same for all the other subdirectories too. That would avoid running so many redundant tests and would lower resource usage.

File: generative_ai/count_token/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+google-cloud-aiplatform[all]==1.68.0
```
Comment (holtskinner) [inline generative_ai/count_token/requirements.txt:1]:
```suggestion
google-cloud-aiplatform[all]==1.69.0
```

File: generative_ai/count_token/test_token_count.py [hunk 2]
```diff
@@ -13,21 +13,34 @@
...
+
+
+def test_gemini_count_simple_example() -> None:
+    response = simple_example.count_tokens_example()
     assert response
     assert response.usage_metadata
 
-    response = multimodal_token_count_example.count_tokens_multimodal()
+
+def test_gemini_multimodal_example() -> None:
```
Comment (holtskinner) [inline generative_ai/count_token/test_token_count.py:39]:
```suggestion
def test_gemini_count_tokens_multimodal_example() -> None:
```

**Review by holtskinner — APPROVED** [review]

File: generative_ai/count_token/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+sentencepiece==0.2.0
```
[the remark this answers is not in the document]
Comment (msampathkumar) [reply generative_ai/count_token/requirements.txt:2]: Yes, this is for tokenizing locally. There are *NO* there instructions in the docs. 

Technically, I see this as missing dependency package requirement in the SDK itself.

**Comment by davidcavazos** [comment]
@msampathkumar thanks for providing the local directory test config. It looks like it's still running everything, not your fault.

After closer inspection at the code that detects the test package directory, it looks like it dives deeper from the repo root directory until it finds a `requirements.txt` file. I would have imagined it would bubble up from the diffs it found, so it'll keep running everything until the `generative_ai/requirements.txt` file is gone.

`btlr run --max-cmd-duration=60m **/requirements.txt --git-diff origin/main... . -- /workspace/.kokoro/tests/run_single_test.sh`

https://github.com/googleapis/btlr/blob/main/cmd/run.go#L325 <- finds most general instead of most specific

**Review by davidcavazos — APPROVED** [review]

**Comment by msampathkumar** [comment]
All count token related tests are passing. Error are related to Embedding and FunctionCalling. Re-running the test

**Comment by msampathkumar** [comment]
Function calling code sample is failing repeatedly for 002 version and so using 001

**Comment by msampathkumar** [comment]
Raising fresh independent PRs

**Comment by msampathkumar** [comment]
https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12638

https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12684

**Pull request closed without merging**